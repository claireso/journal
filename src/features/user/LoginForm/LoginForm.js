import { useRef } from 'react'
import PropTypes from 'prop-types'

import Input from '@components/form/Input'
import SubmitButton from '@components/form/Buttons'

const LoginForm = ({ onSubmit, isProcessing }) => {
  const username = useRef(null)
  const password = useRef(null)

  const handleSubmit = (event) => {
    event.preventDefault()

    if (isProcessing) return

    onSubmit({
      username: username.current.value,
      password: password.current.value
    })
  }

  return (
    <form onSubmit={handleSubmit} method="POST">
      <Input ref={username} autoFocus name="username" label="Username" testId="username" required />
      <Input ref={password} type="password" name="password" label="Password" testId="password" required />
      <SubmitButton value="Log in" isLoading={isProcessing} data-testid="submit" />
    </form>
  )
}

LoginForm.propTypes = {
  onSubmit: PropTypes.func,
  isProcessing: PropTypes.bool
}

LoginForm.defaultProps = {
  onSubmit: () => {},
  isProcessing: false
}

export default LoginForm
