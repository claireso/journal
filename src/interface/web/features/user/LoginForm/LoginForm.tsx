import Input from '@web/components/form/Input'
import ButtonSubmit from '@web/components/form/ButtonSubmit'

interface LoginFormProps {
  action: (payload: FormData) => void
}

const LoginForm = ({ action }: LoginFormProps) => {
  return (
    <form action={action}>
      <Input autoFocus name="username" label="Username" required />
      <Input type="password" name="password" label="Password" required />
      <ButtonSubmit size="lg" block>
        Log in
      </ButtonSubmit>
    </form>
  )
}

export default LoginForm
