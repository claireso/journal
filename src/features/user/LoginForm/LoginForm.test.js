import { render, fireEvent } from '@testing-library/react'

import LoginForm from './LoginForm'

describe('<FormLogin />', () => {
  const defaultProps = {
    onSubmit: () => {},
    isProcessing: false
  }

  const renderComponent = (props = {}) => {
    return render(<LoginForm {...defaultProps} {...props} />)
  }

  test('should render component', () => {
    const { container } = renderComponent()

    expect(container).toMatchSnapshot()
  })

  test('should render component in processing status', () => {
    const { container } = renderComponent({ isProcessing: true })

    expect(container).toMatchSnapshot()
  })

  test('should call onSubmit function', () => {
    const spyOnSubmit = jest.fn()

    const { getByLabelText, getByDisplayValue } = renderComponent({
      onSubmit: spyOnSubmit
    })

    fireEvent.change(getByLabelText(/username/i), {
      target: { value: 'admin' }
    })
    fireEvent.change(getByLabelText(/password/i), {
      target: { value: 'password' }
    })

    fireEvent.click(getByDisplayValue(/log in/i))

    expect(spyOnSubmit).toHaveBeenCalledWith({
      username: 'admin',
      password: 'password'
    })
  })
})
