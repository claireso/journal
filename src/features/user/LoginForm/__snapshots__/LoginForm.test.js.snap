// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormLogin /> should render component 1`] = `
<DocumentFragment>
  .c0 {
  margin: 0 0 var(--gutter);
}

.c1 {
  display: block;
  font-size: 1.6rem;
  line-height: 1.5;
  margin: 0 0 1.2rem;
}

.c2 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: var(--form-input-bg);
  border: 1px solid var(--border-color);
  border-radius: 0.4rem;
  box-shadow: 1px 1px 3px var(--box-shadow);
  display: block;
  font-family: var(--font-family);
  outline: none;
  padding: 0.8rem 1.2rem;
  width: 100%;
  font-size: var(--font-size-normal);
  line-height: 1.71;
  -webkit-transition: border-color 250ms ease-out;
  transition: border-color 250ms ease-out;
}

.c2:focus {
  border-color: var(--border-color-focus);
}

.c3 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: var(--primary-normal);
  border: none;
  border-radius: 3.6rem;
  color: white;
  cursor: pointer;
  display: block;
  font-size: var(--font-size-normal);
  line-height: 1;
  margin: 2.8rem auto 0;
  outline: none;
  padding: 1.6rem 5.2rem;
  -webkit-transition: background 150ms ease-out;
  transition: background 150ms ease-out;
}

.c3:hover {
  background: var(--primary-lighter);
}

<form
    method="POST"
  >
    <div
      class="c0"
    >
      <label
        class="c1"
        for="username"
      >
        Username
      </label>
      <input
        class="c2"
        data-testid="username"
        id="username"
        name="username"
        required=""
        type="text"
        value=""
      />
    </div>
    <div
      class="c0"
    >
      <label
        class="c1"
        for="password"
      >
        Password
      </label>
      <input
        class="c2"
        data-testid="password"
        id="password"
        name="password"
        required=""
        type="password"
        value=""
      />
    </div>
    <input
      class="c3"
      data-testid="submit"
      type="submit"
      value="Log in"
    />
  </form>
</DocumentFragment>
`;

exports[`<FormLogin /> should render component in processing status 1`] = `
<DocumentFragment>
  .c0 {
  margin: 0 0 var(--gutter);
}

.c1 {
  display: block;
  font-size: 1.6rem;
  line-height: 1.5;
  margin: 0 0 1.2rem;
}

.c2 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: var(--form-input-bg);
  border: 1px solid var(--border-color);
  border-radius: 0.4rem;
  box-shadow: 1px 1px 3px var(--box-shadow);
  display: block;
  font-family: var(--font-family);
  outline: none;
  padding: 0.8rem 1.2rem;
  width: 100%;
  font-size: var(--font-size-normal);
  line-height: 1.71;
  -webkit-transition: border-color 250ms ease-out;
  transition: border-color 250ms ease-out;
}

.c2:focus {
  border-color: var(--border-color-focus);
}

.c5 {
  margin: 2rem 0;
  -webkit-animation: iAjNNh 800ms ease-in-out infinite alternate;
  animation: iAjNNh 800ms ease-in-out infinite alternate;
}

.c5:after {
  background: var(--primary-normal);
  border-radius: 50%;
  content: '';
  display: block;
  height: 1rem;
  width: 1rem;
  margin: 0 auto;
}

.c3 {
  margin: 2.8rem 0 0;
  padding: 1.8rem 0 1.8rem;
}

.c3 .c4 {
  margin: 0 auto;
}

<form
    method="POST"
  >
    <div
      class="c0"
    >
      <label
        class="c1"
        for="username"
      >
        Username
      </label>
      <input
        class="c2"
        data-testid="username"
        id="username"
        name="username"
        required=""
        type="text"
        value=""
      />
    </div>
    <div
      class="c0"
    >
      <label
        class="c1"
        for="password"
      >
        Password
      </label>
      <input
        class="c2"
        data-testid="password"
        id="password"
        name="password"
        required=""
        type="password"
        value=""
      />
    </div>
    <div
      class="c3"
    >
      <div
        class="c4 c5"
      />
    </div>
  </form>
</DocumentFragment>
`;
